operator/apis/mlops/v1alpha1/pipeline_types.go

```diff
@@ -25,42 +25,77 @@ import (
 	"github.com/seldonio/seldon-core/apis/go/v2/mlops/scheduler"
 )
 
+//+kubebuilder:object:root=true
+//+kubebuilder:subresource:status
+//+kubebuilder:resource:shortName=mlp
+
+// Pipeline is the Schema for the pipelines API
+type Pipeline struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   PipelineSpec   `json:"spec,omitempty"`
+	Status PipelineStatus `json:"status,omitempty"`
+}
+
+//+kubebuilder:object:root=true
+
+// PipelineList contains a list of Pipeline
+type PipelineList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+
+	Items []Pipeline `json:"items"`
+}
+
 // PipelineSpec defines the desired state of Pipeline
 type PipelineSpec struct {
 	// External inputs to this pipeline, optional
 	Input *PipelineInput `json:"input,omitempty"`
+
 	// The steps of this inference graph pipeline
 	Steps []PipelineStep `json:"steps"`
+
 	// Synchronous output from this pipeline, optional
 	Output *PipelineOutput `json:"output,omitempty"`
 }
 
+// +kubebuilder:validation:Enum=inner;outer;any
 type JoinType string
 
 const (
```

Comment:
**explain:** It's not ideal that it's repeated here, but this approach seems common [1].

[1] https://github.com/search?q=%2Bkubebuilder%3Avalidation%3AEnum&type=code — e.g. Grafana, Cilium, etc.